### PR TITLE
Allow this project to bootstrap itself with tox

### LINKED ISSUE #7: Status code 403 with pip 21.1.x and keyrings.google-artifactregistry-auth 0.0.2
- WSL2/Fedora 33
- Python 3.9.2
- works with pip 21.0.1
- cannot install a package located in my GAR with pip 21.1 or 21.1.1.

```bash
Getting page https://europe-pypi.pkg.dev/xxxxxx-xix9/python/simple/cpe-common/
Found index url https://europe-pypi.pkg.dev/xxxxxx-xix9/python/simple/
Looking up "https://europe-pypi.pkg.dev/xxxxxx-xix9/python/simple/cpe-common/" in the cache
Request header has "max_age" as 0, cache bypassed
Starting new HTTPS connection (1): europe-pypi.pkg.dev:443
https://europe-pypi.pkg.dev:443 "GET /xxxxxx-xix9/python/simple/cpe-common/ HTTP/1.1" 403 0
Status code 403 not in (200, 203, 300, 301)
Could not fetch URL https://europe-pypi.pkg.dev/xxxxxx-xix9/python/simple/cpe-common/: 403 Client Error: Forbidden for url: https://europe-pypi.pkg.dev/xxxxxx-xix9/python/simple/cpe-common/ - skipping
```

```
╰(.venv)─» pip freeze
appdirs==1.4.4
bleach==3.3.0
cachetools==4.2.2
certifi==2020.12.5
cffi==1.14.5
chardet==4.0.0
colorama==0.4.4
cryptography==3.4.7
distlib==0.3.1
docutils==0.17.1
filelock==3.0.12
google-auth==1.30.0
idna==2.10
importlib-metadata==4.0.1
jeepney==0.6.0
keyring==23.0.1
keyrings.google-artifactregistry-auth==0.0.2
packaging==20.9
pkginfo==1.7.0
pluggy==0.13.1
py==1.10.0
pyasn1==0.4.8
pyasn1-modules==0.2.8
pycparser==2.20
Pygments==2.9.0
pyparsing==2.4.7
readme-renderer==29.0
requests==2.25.1
requests-toolbelt==0.9.1
rfc3986==1.4.0
rsa==4.7.2
SecretStorage==3.3.1
six==1.15.0
toml==0.10.2
tox==3.23.0
tqdm==4.60.0
twine==3.4.1
urllib3==1.26.4
virtualenv==20.4.4
webencodings==0.5.1
zipp==3.4.1
```

```bash
╰(.venv)─» keyring --list-backends
keyrings.gauth.GooglePyPIAuth (priority: 9)
keyring.backends.chainer.ChainerBackend (priority: -1)
keyring.backends.fail.Keyring (priority: 0)
```

### PR DESCRIPTION
**Pull request by di**
Fixes #7.

This PR configures this project to double as a [tox plugin](https://tox.readthedocs.io/en/latest/plugins.html). When specified via tox's [`requires`](https://tox.readthedocs.io/en/latest/config.html#conf-requires) configuration option, the package will self-install into the virtual environment before any other dependencies are installed.

In order to ensure that the same version is installed when bootstrapping, the version number for the library is extracted into an importable module (`google_artifactregistry_auth.__version__`).

Also adds tests for the plugin via GitHub Actions, and docs for usage.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,4 +1,6 @@
 import setuptools
 
+import google_artifactregistry_auth
```
Comment (gaborbernat) [inline setup.py:3]: Per PEP-517 there's no guarantee that the cwd will be on the sys path, if you do something like this you'd want to make sure that the parent of ``__file__`` here is on sys path. 

Something like 

https://github.com/python/mypy/blob/master/setup.py#L12-L13

**Review by gaborbernat — APPROVED** [review]

File: setup.py (hunk 1 above)
Comment (di) [reply setup.py]: Ah, great point. There's a better way to do this anyways though: baca1be49f0b743ff17eef98d860fb4edddcff87

**Review by gaborbernat — APPROVED** [review]

**Comment by megan-kuo** [comment]
Thanks for adding these changes!

**Pull request merged**